Tasks/KubernetesV1/src/kubernetes.ts

```diff
@@ -6,69 +6,86 @@ import path = require('path');
+import trm = require('azure-pipelines-task-lib/toolrunner');
+import { getDeploymentMetadata, IsJsonString, getPublishDeploymentRequestUrl, isDeploymentEntity } from 'kubernetes-common-v2/image-metadata-helper';
+import { WebRequest, WebResponse, sendRequest } from 'utility-common-v2/restutilities';
 
-tl.setResourcePath(path.join(__dirname, '..' , 'task.json'));
+tl.setResourcePath(path.join(__dirname, '..', 'task.json'));
 // Change to any specified working directory
 tl.cd(tl.getInput("cwd"));
 
 var registryType = tl.getInput("containerRegistryType", true);
 var command = tl.getInput("command", false);
 const environmentVariableMaximumSize = 32766;
+const addPipelineMetadata = tl.getVariable("ADD_PIPELINE_METADATA");
 
 var kubeconfigfilePath;
 if (command === "logout") {
     kubeconfigfilePath = tl.getVariable("KUBECONFIG");
 }
 // open kubectl connection and run the command
 var connection = new ClusterConnection(kubeconfigfilePath);
-try
-{
-    connection.open().then(  
+try {
+    connection.open().then(
         () => { return run(connection, command) }
     ).then(
-       () =>  {
-           tl.setResult(tl.TaskResult.Succeeded, "");
-           if (command !== "login") {
+        () => {
+            tl.setResult(tl.TaskResult.Succeeded, "");
+            if (command !== "login") {
                 connection.close();
-           }
-       }
-    ).catch((error) => {
-       tl.setResult(tl.TaskResult.Failed, error.message)
-       connection.close();
-    });
+            }
+        }
+        ).catch((error) => {
+            tl.setResult(tl.TaskResult.Failed, error.message)
+            connection.close();
+        });
 }
-catch (error)
-{
+catch (error) {
     tl.setResult(tl.TaskResult.Failed, error.message);
 }
 
-async function run(clusterConnection: ClusterConnection, command: string) 
-{
+async function run(clusterConnection: ClusterConnection, command: string) {
     var secretName = tl.getInput("secretName", false);
     var configMapName = tl.getInput("configMapName", false);
 
-    if(secretName) {
+    if (secretName) {
         await kubectlSecret.run(clusterConnection, secretName);
     }
 
-    if(configMapName) {
+    if (configMapName) {
         await kubectlConfigMap.run(clusterConnection, configMapName);
     }
 
-    if(command) {
+    if (command) {
         await executeKubectlCommand(clusterConnection, command);
-    }  
+    }
+}
+
+function getAllPods(connection: ClusterConnection): trm.IExecSyncResult {
+    const command = connection.createCommand();
+    command.arg('get');
+    command.arg('pods');
+    command.arg(['-o', 'json']);
+    command.arg(getNameSpace());
+    return command.execSync({ silent: true } as trm.IExecOptions);
+}
+
+function getClusterInfo(connection: ClusterConnection): trm.IExecSyncResult {
+    const command = connection.createCommand();
+    command.arg('cluster-info');
+    return command.execSync({ silent: true } as trm.IExecOptions);
 }
```

Comment:
The KubeCtl class defined in kubernetes-common-v2/kubectl-object-model needs to be invoked via its constructor, which requires kubectl path. 
The kubectl task initializes its own cluster connection and never invokes the particular kubectl class defined in the common util file mentioned above. So I am using the existing cluster connection available with this task rather than creating a new instance of KubeCtl class.